File: users-service-jpa/src/main/java/eu/rudisch/users/persistance/model/Role.java
package eu.rudisch.users.persistance.model;

import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.NamedQuery;
import javax.persistence.Table;

/**
 * The persistent class for the role database table.
 * 
 */
@Entity
@Table(name = "role")
@NamedQuery(name = "Role.findAll", query = "SELECT r FROM Role r")
public class Role implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@Column(name = "ID")
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;

	@Column(name = "role")
	private String role;

	// bi-directional many-to-many association to Membership
	@ManyToMany(mappedBy = "roles")
	private Set<Membership> memberships = new HashSet<>();

	public Role() {
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getRole() {
		return this.role;
	}

	public void setRole(String role) {
		this.role = role;
	}

	public Set<Membership> getMemberships() {
		return this.memberships;
	}

	public void setMemberships(Set<Membership> memberships) {
		this.memberships = memberships;
	}

	public Membership addMembership(Membership membership) {
		getMemberships().add(membership);
		return membership;
	}

	public Membership removeMembership(Membership membership) {
		getMemberships().remove(membership);
		return membership;
	}

	@Override
	public String toString() {
		return "Role [id=" + id + ", role=" + role + "]";
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((role == null) ? 0 : role.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Role other = (Role) obj;
		if (role == null) {
			if (other.role != null)
				return false;
		} else if (!role.equals(other.role))
			return false;
		return true;
	}
}

File: authorization-admin-service-spring-kotlin/src/main/kotlin/eu/rudisch/authorization/admin/model/Role.kt
package eu.rudisch.authorization.admin.model

import java.io.Serializable
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.FetchType
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.JoinTable
import javax.persistence.ManyToMany
import javax.persistence.Table

@Entity
@Table(name = "role")
data class Role(
		@Id
		@GeneratedValue(strategy = GenerationType.AUTO)
		val id: Int,

		@Column(name = "name")
		val name: String,

		@ManyToMany(fetch = FetchType.EAGER)
		@JoinTable(name = "permission_role",
				joinColumns = [JoinColumn(name = "role_id", referencedColumnName = "id")],
				inverseJoinColumns = [JoinColumn(name = "permission_id", referencedColumnName = "id")])
		val permissions: MutableList<Permission>
) : Serializable {
}